bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyDeviceHandler.java

```diff
@@ -85,14 +85,19 @@ public HarmonyDeviceHandler(Thing thing, HarmonyHubHandlerFactory factory) {
     public void handleCommand(ChannelUID channelUID, Command command) {
         logger.trace("Handling command '{}' for {}", command, channelUID);
 
-        Channel channel = getThing().getChannel(channelUID.getId());
-        if (channel == null) {
-            logger.warn("No such channel: {}", channelUID);
+        if (command instanceof RefreshType) {
+            // nothing to refresh
             return;
         }
 
-        if (command instanceof RefreshType) {
-            // nothing to refresh
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.warn("Hub is offline, ignoring command {} for channel {}", command, channelUID);
+            return;
+        }
+
+        Channel channel = getThing().getChannel(channelUID.getId());
```

Comment:
Why did you add this here? You should assume that the framework makes sure that commands are only sent for existing channels.

Reply:
My guess is that i was using that channel variable at some point during testing, and needed to check it's null-ness to appease the nullable annotations, but since i'm not using it for anything, its very much not necessary.